### DISCUSSION TITLE
Some questions to utils.py settings

### DISCUSSION THREAD

**Discussion by Becker884** [Louisvdw/dbus-serialbattery#423] [Q&A] [answered]
Hi,
can you tell me what the following settings do:
![grafik](https://user-images.githubusercontent.com/81651521/212664580-4218367d-8e2f-433f-9ac7-3db955013d3e.png)
1. for what / when will be float voltage used ?
is absorption time only 15min ? and after 15 min CVL goes to float voltage?

2. for what is CVCM used ? it´s default disabled.

greetings

**Comment by Louisvdw** [comment]
Hi @Becker884 
Some users want to lower their charge voltage when their battery has been fully charged. This mimmics the float charging state that some smart chargers use for lead acid batteries. 
For litium batteries this is not required. The BMS handles the cells much better (something that lead acid batteries did not have, and the reason the float stage was added). So it is optional available, but disabled by default.

**Reply by Becker884** [reply]
thank you, so CVCM_Enable refers to the min/max/float values marked above? (all yellow) 
I was a little confused as the two are so far apart.

**Comment by Becker884** [comment]
I´ve a new question to float voltage: how long will the MP2 hold float voltage (3,45V) ? 
infinity or or as long as in VE Configure set ? (with: CVCM_ENABLE = False)

**Reply by mr-manuel** [reply, marked as answer]
The v0.14.3 has a little bug in that. If you are using step mode then `MAX_VOLTAGE_TIME_SEC` lowers the voltage to float after max voltage was reached for this time period.
